src/GafferScene/IECoreGLPreview/Renderer.cpp

```diff
@@ -1099,16 +1100,28 @@ class OpenGLRenderer final : public IECoreScenePreview::Renderer
 		{
 			const bool selected = parameter<bool>( parameters, "selection", false );
 
+			const PathMatcherData *omitted = nullptr;
+			IECore::CompoundDataMap::const_iterator it = parameters.find( "omitted" );
+			if( it != parameters.end() )
+			{
+				omitted = reportedCast<const PathMatcherData>( it->second.get(), "parameter", "omitted" );
+			}
+
 			processQueue();
 			removeDeletedObjects();
 
 			Box3f result;
 			for( const auto &o : m_objects )
 			{
-				if( selected && !o->selected( m_selection ) )
+
+				if(
+					( selected && !o->selected( m_selection ) ) ||
+					( omitted && ( omitted->readable().match( o->name() ) & PathMatcher::ExactMatch ) )
```

Comment:
I guess the quick `if omitted` check here might be one reason not to use `parameter()` above? But then we have the slight overhead of `readable()` as well. Maybe this?

```
const PathMatcher omitted = parameter<PathMatcher( ... );
const bool omittedEmpty = omitted.isEmpty();
...
if( !omittedEmpty && omitted.match() ... )
```

Maybe none of this matters - it's not like I've measured the cost of each approach, but there is at least a bit less code in this version.

Reply:
Should we also be omitting for ancestor matches here though? I think we probably would want to in the case of something parented under the light?